---
####################################################################
## Ulimit Config
####################################################################

- block:

  - name: "Check if {{default_ulimits_dir}} directory exists (may not be available on SuSe)"
    stat:
      path: "{{default_ulimits_dir}}"
    check_mode: no
    register: limitsd_dir
  
  - debug: var=limitsd_dir

  - set_fact:
      ulimit_file: "{% if limitsd_dir.stat.exists %}{{ulimit_file_loc}}{% else %}{{suse_ulimit_file_loc}}{% endif %}"

  - debug: var=ulimit_file

  - debug:
      msg: "Did not find {{default_ulimits_dir}} directory. Configuring ulimits in /etc/security location."
    when: not limitsd_dir.stat.exists
    
  - name: Ensure customer ulimit conf file exists
    copy:
      content: "#Custom file for Viya 3.4 ulimits#\n"
      dest: "{{ulimit_file}}"
      force: no
      group: root
      owner: root
      mode: 0644

  - name: "Setting ulimits for Viya - in {{ulimit_file}}"
    become: yes
    become_user: root
    pam_limits:
      dest: "{{ulimit_file}}"
      domain: "{{item.domain}}"
      limit_type: "{{item.type}}"
      limit_item: "{{item.item}}"
      value: "{{item.value}}"
      use_max: "{{item.use_max}}"
      use_min: "{{item.use_min}}"
      ## certain versions of ansible have a weird output with comments. (see https://github.com/ansible/ansible/issues/27635)
      #comment: "{{item.comment}}"
      backup: yes
    with_items:
      - "{{ulimits}}"

## TODO: add a check to confirm whether {{sas_user}} exists yet

  - name: Gather the current Ulimit values for user {{sas_user}} and root
    become: yes
    become_user: "{{item}}"
    shell: ulimit -a
    changed_when: False
    check_mode: no
    register: review_ulimits
    ignore_errors: true
    with_items:
      - "{{sas_user}}"
      - root
  - debug: var=review_ulimits

  ## block end
  tags:
    - ulimit_config
